Common/Tools/BuildTasks/ExtractLambdasFromXaml.cs

```diff
@@ -113,7 +113,7 @@ private struct LambdaInfo {
                 return GenerateOutput();
             } catch (Exception ex) {
                 LogError(null, 0, 0, ex.Message);
-                return false;
+                throw;
```

Comment:
Roslyn says to be more specific or to rethrow. Since this is used in the build I choose to log and rethrow and this avoids bringing higher level projects into the buildtask.proj

Reply:
I would prefer it if the code logged specific exceptions (IsCriticalException(...)) and returned false. If there is an exception not caught in the specific exception list, then the exception can propagate up the call stack. This is consistent with existing behavior of other code in the file.

Reply:
It's okay to throw when implementing Execute.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.framework.itask.execute?view=netframework-4.8#Microsoft_Build_Framework_ITask_Execute

Reply:
Yup.. just did a test. changed the task to throw.